mdata/store_mock.go

```diff
@@ -0,0 +1,48 @@
+package mdata
+
+import "github.com/raintank/metrictank/mdata/chunk"
+
+type mockSearchResult struct {
+	chunks []chunk.IterGen
+	err    error
+}
+
+type mockStore struct {
```

Comment:
can you explain this a bit more. how does this work, what is CurrCall for, how/when should we use AddMockResult and ResetMock and why, what does Search return (on a high level)?